### PR TITLE
Data validation for citation

### PR DESCRIPTION
**Pull request by klai95**
Per the [stack trace](https://app.datadoghq.com/apm/error-tracking?query=&issueId=ecb2d880-442e-11ed-89d9-da7ad0900002&start=1664985275287&end=1664988875287&paused=false) the line that caused the issue is: https://github.com/chanzuckerberg/napari-hub/blob/main/backend/utils/github.py#L170. This PR will handle the data validation for the issue and ensure Datadog no longer contains errors.

### CODE REVIEW

File: backend/utils/github.py [hunk 1]
```diff
@@ -201,6 +204,8 @@ def get_citation_author(citation_str: str) -> Union[Dict[str, str], None]:
     :return: list of mappings between the string 'name' and the author name
     """
     citation_yaml = yaml.safe_load(citation_str)
+    if citation_yaml is None:
+        return {}
```
Comment (richaagarwal) [inline backend/utils/github.py:208]: @klai95 I think this should be `return []` since this function is returning a list of authors or not.

Comment (klai95) [reply backend/utils/github.py]: Yup made the change thanks for the catch

File: backend/utils/github.py [hunk 2]
```diff
@@ -200,7 +203,10 @@ def get_citation_author(citation_str: str) -> Union[Dict[str, str], None]:
     :param citation_str: citation string to parse
     :return: list of mappings between the string 'name' and the author name
     """
-    citation_yaml = yaml.safe_load(citation_str)
+    try:
+        citation_yaml = yaml.safe_load(citation_str)
```
[the remark this answers is not in the document]
Comment (klai95) [reply backend/utils/github.py:207]: I think we should wrap all the code in try and below it include the except clause right?

[the remark this answers is not in the document]
Comment (richaagarwal) [reply backend/utils/github.py:207]: Yeah, I think you need to add a `return []` to that. I think that's preferable to moving more code inside the try/except block.

[the remark this answers is not in the document]
Comment (klai95) [reply backend/utils/github.py:207]: I don't see any errors as of now. I will monitor it to ensure everything works

File: backend/utils/github.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import json
+import logging
 import os.path
 import re
+import traceback
```
Comment (richaagarwal) [inline backend/utils/github.py:5]: Let's remove this now unused import

Comment (klai95) [reply backend/utils/github.py]: Thanks for the catch, just made the change!

**Review by richaagarwal — APPROVED** [review]

**Pull request merged**